app/src/main/java/com/github/sdp/ratemyepfl/database/ItemsRepository.kt

```diff
@@ -35,12 +29,37 @@ class ItemsRepository @Inject constructor() : ItemsRepositoryInterface, Reposito
             .mapNotNull { obj -> obj.toClassroom() }
     }
 
-    suspend fun getByIdClassrooms(id: String): Classroom? {
+    override suspend fun getByIdClassrooms(id: String): Classroom? {
         return getById(collectionRooms, id).toClassroom()
     }
 
     override suspend fun getRestaurants(): List<Restaurant?> {
         return getLimit(collectionRestaurants, 50)
             .mapNotNull { obj -> obj.toRestaurant() }
     }
+
+    override suspend fun getByIdRestaurants(id: String): Restaurant? {
+        return getById(collectionRestaurants, id).toRestaurant()
+    }
+
+    // Function to get for a generic Reviewable
+    override suspend fun getById(id : String?): Reviewable? {
```

Comment:
I'll leave a global comment for the team at the end of the PR, but I think that we should avoid using nullable values as much as possible. For example, here if the value is null you return null. Why do you not forbid id to be null? It would be the caller's responsibility to verify it (the ? operator makes it very easy in Kotlin)

Reply:
Then we can fake the DB as we want, i.e returns something even if id is null. Without the nullable the tests will crash if not given an id as extra.

Reply:
I agree on this point, but we should not optimize for the tests ;) I mean you are still using a fake database, isn't it? So the id does not play any role in the test, and write null is longer than "".

Reply:
It's a tradeoff, just one check added in the repository function or forget the rule for tests and start using the scenarios all over again . I personally prefer the first option.

Reply:
Feel free to do it since the repo is not refactored yet. Anyway, there's something I don't like with the structure of the repositories. We have many tools (genericity, sealed class), and we should use them. I'll deeply think about it for the next meeting.